tests/Console/ConfigurationResolverTest.php

```diff
@@ -246,11 +246,53 @@ public function testResolveConfigFileChooseFileWithInvalidFile()
     public function testResolvePathRelative()
     {
         $this->resolver
-            ->setCwd(__DIR__)
-            ->setOption('path', 'Foo'.DIRECTORY_SEPARATOR.'Bar')
+            ->setCwd(dirname(__DIR__))
+            ->setOption('path', basename(__DIR__))
             ->resolve();
 
-        $this->assertSame(__DIR__.DIRECTORY_SEPARATOR.'Foo'.DIRECTORY_SEPARATOR.'Bar', $this->resolver->getPath());
+        $this->assertSame(__DIR__, $this->resolver->getPath());
+    }
+
+    public function testResolvePathWithFileThatIsExcludedDirectly()
+    {
+        $dir = realpath(__DIR__);
+        $this->config->getFinder()
+            ->in($dir)
+            ->notPath(substr(__FILE__, strlen($dir) + 1));
+
+        $this->resolver
+            ->setOption('path', __FILE__)
+            ->resolve();
+
+        $this->assertCount(0, iterator_to_array($this->resolver->getConfig()->getFinder()));
```

Comment:
but I doesn't handle iterators well on all versions (https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/1632#issuecomment-167264865), we could try this case out though :)